lib/ma_crossover/events/life_stop.js

```diff
@@ -8,6 +8,14 @@
  * @param {AOInstance} instance - AO instance
  * @returns {Promise} p - resolves on completion
  */
-const onLifeStop = async (instance = {}) => {}
+const onLifeStop = async (instance = {}) => {
+  const { state = {}, h = {} } = instance
+  const { orders = {}, gid } = state
+  const { emit, debug } = h
+
+  debug('detected ma crossover algo cancelation, stopping...')
+
+  await emit('exec:order:cancel:all', gid, orders)
```

Comment:
take care with that, ma crossover is a special case...

Reply:
https://github.com/bitfinexcom/bfx-hf-algo/blob/79f857bb333387d38a13350bcc9709bc91abea9d/lib/ma_crossover/events/data_managed_candles.js#L92-L102

Reply:
when a cross happens, it will submit an order and exit with `exec:stop`. 

in case the user selected limit orders, we would cancel it

Reply:
maybe you can pass a parameter via the event system to `onLifeStop`

```js
return emit('exec:stop', { keepOrdersOpen: true })
```

Reply:
used this to keep orders alive instead of using `updateState`.. thanks!!